### ISSUE TITLE
Replace Granite.Services.Settings with GLib.Settings

### ISSUE THREAD

**Issue by danrabbit** [elementary/music#550]

**Comment by danielopitz** [comment]
It seems Music uses two different settings managers. `src/Application.vala:31` is already a `GLib.Settings` instance, but there is a `Granite.Services.Settings` subclass in `core/Settings.vala` referenced in 20 places in `core` and src. 

How would you like it to be done? Should everything use the already available `Music.App.settings` (`GLib.Settings` instance at `src/Application.vala:31`)?

And a separate question: Any idea why the code is divided between `src/` and `core/`? Was / is it some kind of convention? Several other Elementary apps I checked didn't had this distinction.

**Comment by danrabbit** [comment]
Yes it would be best to make use of the existing `Music.App.settings` instance

The pieces in "Core" are actually meant to be things necessary for the library used by the plugin interface. At the moment, there are no known 3rd party plugins, so combining `src/` and `core/` would probably be an improvement

This is a pretty old and messy code base. Good luck! :)

**Comment by danielopitz** [comment]
OK, so for the starters I'll move the codebase to use `GLib.Settings`. Merging core with src is a larger refactoring and I would rather do it in a separate PR.

**Comment by danielopitz** [comment]
Well, it's a bummer, but it seems to be impossible to access Music.App from the `core/` directory with the way the code is structured now. Or, infinitely more probable, I just don't know enough about Vala yet. Any ideas? :)

If it is not possible, I would give a shot on merging both directories and doing this change afterwards.

**Comment by danrabbit** [comment]
Right that's correct because `core/` represents the library and Music consumes that library. It would probably make sense to merge things first and then port to GLib.Settings in a separate branch after :+1:

**Comment by danielopitz** [comment]
Ok, I'll try my best with it :)

**Comment by danielopitz** [comment]
So I moved `core/` and `plugins/` to `src/` and updated the build file to make it compile. It compiles just fine, but when I want to start it, it dies right away and I basically hit a roadblock at this point. 

```
(io.elementary.music:16277): GLib-GIO-CRITICAL **: 00:03:38.257: g_application_run: assertion 'G_IS_APPLICATION (application)' failed
```
is what I'm left with. I looked around and it seems that something goes wrong in the constructor, because the line it fails on is:
`src/Application.vala`
```
public static int main (string[] args) {
    Gtk.init (ref args);
    Gda.init ();

    try {
        Gst.init_check (ref args);
    } catch (Error err) {
        error ("Could not init GStreamer: %s", err.message);
    }

    GLib.Environ.set_variable ({"PULSE_PROP_media.role"}, "audio", "true");

    var app = new Music.App ();
    return app.run (args); <-- HERE, app is NULL here
}
```
I stepped through with gdb and it is definitely NULL here.

My problem is, I have no idea why. Constructor blocks don't seem to be called at all (debug messages not displayed, G_MESSAGES_DEBUG=all is set). If the error would be further down the line I would perhaps manage, but I have no clue whatsoever why `app` is NULL in this case.

Any help / idea? 

P.S.:
I never used a Debian-family distro before and I'm having troubles getting debug symbols / sources for all the dependencies installed.
